File: Android/dokit-weex/src/main/java/com/didichuxing/doraemonkit/weex/info/WeexInfoAdapter.java
package com.didichuxing.doraemonkit.weex.info;

import android.content.Context;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.didichuxing.doraemonkit.weex.R;

import java.util.List;

/**
 * @author haojianglong
 * @date 2019-06-25
 */
public class WeexInfoAdapter extends BaseAdapter {

    private Context mContext;

    private List<WeexInfo> mWeexInfos;

    public void setWeexInfos(List<WeexInfo> infos) {
        this.mWeexInfos = infos;
    }

    public WeexInfoAdapter(Context context) {
        this.mContext = context;
    }

    @Override
    public int getCount() {
        return mWeexInfos.size();
    }

    @Override
    public Object getItem(int position) {
        return mWeexInfos.get(position);
    }

    @Override
    public long getItemId(int position) {
        return 0;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        if (convertView == null) {
            convertView = LayoutInflater.from(mContext)
                    .inflate(R.layout.dk_fragment_info_item, null);
        }
        ViewHolder viewHolder;
        if (convertView.getTag() == null) {
            viewHolder = new ViewHolder();
            viewHolder.keyText = convertView.findViewById(R.id.info_item_key);
            viewHolder.valueText = convertView.findViewById(R.id.info_item_value);
            convertView.setTag(viewHolder);
        } else {
            viewHolder = (ViewHolder) convertView.getTag();
        }
        setText(viewHolder.keyText, mWeexInfos.get(position).key);
        setText(viewHolder.valueText, mWeexInfos.get(position).value);
        convertView.setLayoutParams(new RelativeLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT,
                (int) mContext.getResources().getDimension(R.dimen.dk_title_height)));
        return convertView;
    }

    private void setText(TextView textView, String text) {
        if (TextUtils.isEmpty(text)) {
            textView.setVisibility(View.GONE);
        } else {
            textView.setVisibility(View.VISIBLE);
            textView.setText(text);
        }
    }

    class ViewHolder {
        TextView keyText;
        TextView valueText;
    }

}

File: Android/kotlin/doraemonkit-weex/src/main/java/com/didichuxing/doraemonkit/weex/info/adapter/WeexInfoAdapter.kt
package com.didichuxing.doraemonkit.weex.info.adapter

import android.content.Context
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.RelativeLayout
import android.widget.TextView
import com.didichuxing.doraemonkit.weex.R
import com.didichuxing.doraemonkit.weex.info.WeexInfo
import com.didichuxing.doraemonkit.weex.util.getDimensionPixel

/**
 * Transformed by alvince on 2020/6/30
 *
 * @author haojianglong
 * @date 2019-06-25
 */
class WeexInfoAdapter(private val context: Context) : BaseAdapter() {

    private val weexInfoList = mutableListOf<WeexInfo>()

    override fun getItem(position: Int): WeexInfo? = position
        .takeIf { it in 0 until count }
        ?.let { weexInfoList[it] }

    override fun getItemId(position: Int): Long = 0L

    override fun getCount(): Int = weexInfoList.size

    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
        val view = convertView
            ?: LayoutInflater.from(context).inflate(R.layout.dk_fragment_info_item, null)
        return view.also { v ->
            val holder = v.tag
                ?.let { it as? ViewHolder }
                ?: ViewHolder().apply {
                    keyText = v.findViewById(R.id.info_item_key)
                    valueText = v.findViewById(R.id.info_item_value)
                    v.tag = this
                }
            holder.apply {
                getItem(position)?.also { info ->
                    keyText?.bind(info.key)
                    valueText?.bind(info.value)
                }
                v.layoutParams = RelativeLayout.LayoutParams(
                    ViewGroup.LayoutParams.MATCH_PARENT,
                    context.getDimensionPixel(R.dimen.dk_title_height)
                )
            }
        }
    }

    fun setWeexInfos(list: List<WeexInfo>) {
        weexInfoList.apply {
            clear()
            addAll(list)
        }
    }

    private fun TextView.bind(text: String?) {
        (text ?: "").also { t ->
            if (TextUtils.isEmpty(text)) {
                visibility = View.GONE
            } else {
                visibility = View.VISIBLE
                this.text = t
            }
        }
    }


    internal class ViewHolder {
        var keyText: TextView? = null
        var valueText: TextView? = null
    }

}